### PR TITLE
SAM-3091:Can't delete an assessment with no visible submissions

### PR DESCRIPTION
**Pull request by raulsv**

### CODE REVIEW

File: samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/assessment/PublishedAssessmentService.java [hunk 1]
```diff
@@ -341,10 +341,10 @@ public Integer getTotalSubmission(String agentId, String publishedAssessmentId)
    * @param publishedAssessmentId
    * @return number of submissions
    */
-  public Integer getTotalSubmissionForEachAssessment(String publishedAssessmentId){
+  public Integer getTotalSubmissionForEachAssessment(String publishedAssessmentId, Set usersAbleCreateAssessments){
```
Comment (ern) [inline samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/assessment/PublishedAssessmentService.java]: use Generics

File: samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/PublishedAssessmentFacadeQueries.java [hunk 2]
```diff
@@ -1503,14 +1503,27 @@ public PublishedItemText loadPublishedItemText(Long itemTextId) {
...
      * @return number of submissions
      */
-	public Integer getTotalSubmissionForEachAssessment(final Long publishedAssessmentId) {
-		final HibernateCallback<List<Number>> hcb = session -> session
-				.createQuery("select count(a) from AssessmentGradingData a where a.forGrade = :forgrade and a.publishedAssessmentId = :id")
-				.setBoolean("forgrade", true)
-				.setLong("id", publishedAssessmentId)
-				.list();
-		List<Number> l = getHibernateTemplate().execute(hcb);
+	public Integer getTotalSubmissionForEachAssessment(final Long publishedAssessmentId, Set usersAbleCreateAssessments) {
```
Comment (ern) [inline samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/PublishedAssessmentFacadeQueries.java]: Use Generics

File: samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/PublishedAssessmentFacadeQueries.java [hunk 3]
```diff
@@ -1503,14 +1503,27 @@ public PublishedItemText loadPublishedItemText(Long itemTextId) {
...
+		final String query = "select count(a) from AssessmentGradingData a where a.forGrade = :forgrade and a.publishedAssessmentId = :id";
+ 		final HibernateCallback<List<Number>> hcb;
+		
+		if (usersAbleCreateAssessments!=null && usersAbleCreateAssessments.size()>0){
```
Comment (ern) [inline samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/PublishedAssessmentFacadeQueries.java]: Be nice if used Criterion api

**Comment by raulsv** [comment]
We are having a discussion on what is the best solution to resolve this bug in https://jira.sakaiproject.org/browse/SAM-3091. Therefore I close this PR.

**Pull request closed without merging**